src/ServerConfig.chpl

```diff
@@ -291,17 +307,43 @@ module ServerConfig
                                          (getMemLimit():real * numLocales)) * 100):uint));
                 }
             }
-            if total > getMemLimit() {
-                var pct = AutoMath.round((total:real / getMemLimit():real * 100):uint);
-                var msg = "cmd requiring %i bytes of memory exceeds %i limit with projected pct memory used of %i%%".format(
+            
+            /*
+             * If the MemoryMgmt.memMgmtType is STATIC (default), use the memory management logic based upon
+             * a percentage of the locale0 host machine physical memory. 
+             *
+             * If DYNAMIC, use the new dynamic memory mgmt capability in the MemoryMgmt module that first determines 
+             * for each locale if there's sufficient space within the memory currently allocated to the Arkouda 
+             * Chapel process to accommodate the projected memory required by the cmd. If not, then MemoryMgmt 
+             * checks the available memory on each locale to see if more can be allocated to the Arkouda-Chapel process.
+             * If the answer is no on any locale, the cmd is not executed and MemoryMgmt logs the corresponding locales
+             * server-side. More detailed client-side reporting can be implemented in a later version. 
+             */
+            if memMgmtType == MemMgmtType.STATIC {
+                if total > getMemLimit() {
+                    var pct = AutoMath.round((total:real / getMemLimit():real * 100):uint);
+                    var msg = "cmd requiring %i bytes of memory exceeds %i limit with projected pct memory used of %i%%".format(
                                    total, getMemLimit(), pct);
-                scLogger.error(getModuleName(),getRoutineName(),getLineNumber(), msg);  
-                throw getErrorWithContext(
-                          msg=msg,
-                          lineNumber=getLineNumber(),
-                          routineName=getRoutineName(),
-                          moduleName=getModuleName(),
-                          errorClass="ErrorWithContext");                                        
+                    scLogger.error(getModuleName(),getRoutineName(),getLineNumber(), msg);  
+                    throw getErrorWithContext(
+                              msg=msg,
+                              lineNumber=getLineNumber(),
+                              routineName=getRoutineName(),
+                              moduleName=getModuleName(),
+                              errorClass="ErrorWithContext");                                        
```

Comment:
wouldn't it make sense to call `checkStaticMemoryLimit` here or am i misreading

Reply:
ROTFL, why yes, yes it would

Reply:
@pierce314159 I actually ripped the checkStaticMemoryLimit outta there...really did not help readability.

Reply:
it looks like it's still there to me, idk if the changes didn't make their way to github somehow. But this is also something that can be addressed in a future cleanup deal.